x-pack/plugins/uptime/server/lib/synthetics_service/synthetics_service.ts

```diff
@@ -294,31 +298,57 @@ export class SyntheticsService {
 
   async getMonitorConfigs() {
     const savedObjectsClient = this.server.savedObjectsClient;
+    const encryptedClient = this.server.encryptedSavedObjects.getClient();
+    const monitors: Array<SavedObject<SyntheticsMonitorWithSecrets>> = [];
 
     if (!savedObjectsClient?.find) {
       return [] as SyntheticsMonitorWithId[];
     }
 
-    const findResult = await savedObjectsClient.find<SyntheticsMonitor>({
+    const { saved_objects: encryptedMonitors } = await savedObjectsClient.find<SyntheticsMonitor>({
       type: syntheticsMonitorType,
       namespaces: ['*'],
-      perPage: 10000,
+      perPage: 500,
+    });
+
+    const start = Date.now();
+
+    for (const monitor of encryptedMonitors) {
+      const decryptedMonitor =
+        await encryptedClient.getDecryptedAsInternalUser<SyntheticsMonitorWithSecrets>(
+          syntheticsMonitor.name,
+          monitor.id,
+          {
+            namespace: monitor.namespaces?.[0],
+          }
+        );
+      monitors.push(decryptedMonitor);
+    }
+
+    const end = Date.now();
+    const duration = end - start;
+
+    this.logger.info(`Decrypted ${monitors.length} monitors. Took ${duration} miliseconds`, {
```

Comment:
i am interested on the severity, may be add a threshold, if it takes too long, add as an info log, so that it can be surfaced easily, otherwise keep as debug log. WDYT?